process/interface.go

```diff
@@ -41,6 +41,30 @@ type HeaderValidator interface {
 	IsHeaderValidForProcessing(headerHandler data.HeaderHandler) bool
 }
 
+// InterceptedDataFactory can create new instances of InterceptedData
+type InterceptedDataFactory interface {
+	Create(buff []byte) (InterceptedData, error)
+}
+
+// InterceptedData represents the interceptor's view of the received data
+type InterceptedData interface {
+	CheckValid() error
```

Comment:
CheckValidity

Reply:
Changed. CheckValidity was not called in any of the new interceptors. Added calls + test